components/compliance-service/reporting/relaxting/reports.go

```diff
@@ -136,178 +109,71 @@ func (backend ES2Backend) getNodeReportIdsFromTimeseries(esIndex string,
+		// Contains(filteredReportIds, reportId) call
+		reportNode := make(map[string]string, len(nodeReport))
+		for nodeID, reportID := range nodeReport {
+			reportNode[reportID] = nodeID
+		}
 
-	minDate, found := searchResult.Aggregations.Min("min_date")
-	if !found {
-		return &complianceBirthday, errors.Wrap(err, "GetMinScanDate unable to read min_date")
+		// filtering out the nodes for which the report id is no longer in filteredReportIds
+		filteredNodeReport := make(map[string]string, len(filteredReportIds))
+		for _, reportID := range filteredReportIds {
+			filteredNodeReport[reportNode[reportID]] = reportID
+		}
+		nodeReport = filteredNodeReport
 	}
 
-	var dateAsString string
-	err = json.Unmarshal(*minDate.Aggregations["value_as_string"], &dateAsString)
-	if err != nil {
-		return &complianceBirthday, errors.Wrap(err, "GetMinScanDate unable to Unmarshal min_date")
-	}
+	reportIds := MapValues(nodeReport)
 
-	logrus.Debugf("GetMinScanDate: earliest scan date = %s", dateAsString)
+	logrus.Debugf("getNodeReportIdsFromTimeseries returning %d report ids in %d milliseconds\n",
+		len(reportIds), searchResult.TookInMillis)
 
-	formatOfDate := "2006-01-02"
-	earliestScanDate, err := time.Parse(formatOfDate, dateAsString)
-	if err != nil {
-		return &complianceBirthday, errors.Wrap(err, "GetMinScanDate unable to Parse min_date")
-	}
-	return &earliestScanDate, nil
+	return reportIds, nil
 }
 
-//GetMaxScanDate gets the date of the newest scan in ES
-func (backend ES2Backend) GetMaxScanDate() (*time.Time, error) {
-	complianceBirthday := time.Date(2017, time.January, 1, 0, 0, 0, 0, time.UTC)
-	aggs := elastic.NewMaxAggregation().Field("end_time").Format("yyyy-MM-dd")
-	searchSource := elastic.NewSearchSource().
-		Aggregation("max_date", aggs).
-		Size(0)
-
-	esIndex := ComplianceDailySumTwenty
-	source, err := searchSource.Source()
-	if err != nil {
-		return &complianceBirthday, errors.Wrap(err, "GetMaxScanDate unable to get Source")
-	}
-	LogQueryPartMin(esIndex, source, "GetMaxScanDate query searchSource")
-
-	client, err := backend.ES2Client()
-	if err != nil {
-		return &complianceBirthday, errors.Wrap(err, "GetMaxScanDate cannot connect to ElasticSearch")
-	}
-
-	searchResult, err := client.Search().SearchSource(searchSource).
-		Index(ComplianceDailySumTwenty).
-		FilterPath("aggregations.max_date").
-		Do(context.Background())
-
-	if err != nil {
-		return &complianceBirthday, errors.Wrap(err, "GetMaxScanDate unable to complete search")
-	}
-
-	LogQueryPartMin(esIndex, searchResult.Aggregations, "GetMaxScanDate - search results aggs")
-
-	maxDate, found := searchResult.Aggregations.Min("max_date")
-	if !found {
-		return &complianceBirthday, errors.Wrap(err, "GetMaxScanDate unable to read max_date")
-	}
-
-	var dateAsString string
-	err = json.Unmarshal(*maxDate.Aggregations["value_as_string"], &dateAsString)
+func (backend ES2Backend) GetReportIds(esIndex string, filters map[string][]string) ([]string, error) {
```

Comment:
The diff got confused here, but this function was simplified by removing an unnecessary intermediary function (`GetNodeReportIds`) and by changing the return value of `getNodeReportIdsFromTimeseries`.